bin/metric-postgres-locks.rb

```diff
@@ -77,33 +80,38 @@ class PostgresStatsDBMetrics < Sensu::Plugin::Metric::CLI::Graphite
          default: nil
 
   include Pgpass
+  include Pgdatabases
 
   def run
     timestamp = Time.now.to_i
 
     locks_per_type = Hash.new(0)
     pgpass
-    con     = PG.connect(host: config[:hostname],
-                         dbname: config[:database],
-                         user: config[:user],
-                         password: config[:password],
-                         port: config[:port],
-                         connect_timeout: config[:timeout])
-    request = [
-      'SELECT mode, count(mode) AS count FROM pg_locks',
-      "WHERE database = (SELECT oid FROM pg_database WHERE datname = '#{config[:database]}')",
-      'GROUP BY mode'
-    ]
-
-    con.exec(request.join(' ')) do |result|
-      result.each do |row|
-        lock_name = row['mode'].downcase.to_sym
-        locks_per_type[lock_name] = row['count']
+    databases = pgdatabases
+    con       = PG.connect(host: config[:hostname],
+                           dbname: databases.first,
```

Comment:
why only the first database?

Reply:
The first database from the list is simply used to make a valid connection and the locks statistics are then queried through SQL statements for all databases. It is simply not necessary to make individual connections for each database. Please see line 99-103 for the queries run for each database.

Reply:
ah gotcha, that makes sense, would not hurt to add a comment about that. I will review this in greater detail when I have more time. Is there any way we can provide backwards compatibility?

Reply:
I currently see two possible options:

Option 1:
Require the user to set the `--db` option to `*` if all databases should be checked. If it is not set to `*` the current behaviour is retained.
But in `lib/sensu-plugins-postgres/pgpass.rb` there is a check for `*` to switch to fallbacks and to be consistent the switch to fallbacks should no longer be done which will also introduce a breaking change (with less users affected).

Option 2:
Add a new option e.g. `--all-databases` which will override the `--db` option. This means the user has to specify the `--all-databases` option if he really would like to get metrics for all databases or use the existing `--db` option to list database names explicitly.

Also Option 1 has the advantage over Option 2 that it is easier to parameterize the check through tokens if there is no option to invalidate/override another.

Let me know what you think or if you have another idea.

Reply:
I think its a trade off between backwards compatibility and code readability I tend to prefer to not break users regardless of how small the numbers are unless there is a compelling reason.